pkg/operator/controllers/imageconfig/image_controller.go

```diff
@@ -119,18 +119,26 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // Switch case to ensure the correct registries are added depending on the cloud environment (Gov or Public cloud)
 func getCloudAwareRegistries(instance *arov1alpha1.Cluster) ([]string, error) {
-	var requiredRegistries []string
+	const replicationRegistryFormat = "%s.%s.data.%s"
+	var replicationRegistry string
+	var dnsSuffix string
+
+	acrDomain := instance.Spec.ACRDomain
+	acrSubdomain := strings.Split(acrDomain, ".")[0]
+	if len(acrDomain) == 0 || len(acrSubdomain) == 0 {
```

Comment:
nit: comparing a string to an empty string is a bit more idomatic than checking it's length.  

```suggestion
	if acrDomain == "" || len(acrSubdomain) == 0 {
```